calico-vpp-agent/routing/connectivity/ipip.go

```diff
@@ -57,6 +59,26 @@ func (p *IpipProvider) RescanState() {
 			p.ipipIfs[tunnel.Dst.String()] = tunnel
 		}
 	}
+
+	p.log.Infof("Rescanning existing routes")
+	p.ipipRoutes = make(map[uint32]map[string]bool)
+	routes, err := p.vpp.GetRoutes(0, false)
+	if err != nil {
+		p.log.Errorf("Error listing routes: %v", err)
+	}
+	for _, route := range routes {
+		for _, routePath := range route.Paths{
+			for _, tunnel := range p.ipipIfs{
+				if routePath.SwIfIndex == tunnel.SwIfIndex{
+					_, found := p.ipipRoutes[tunnel.SwIfIndex]
+					if !found {
+						p.ipipRoutes[tunnel.SwIfIndex] = make(map[string]bool)
+					}
+					p.ipipRoutes[tunnel.SwIfIndex][route.String()] = true
```

Comment:
`route.Dst.String()` rather no ?

Reply:
I was hesitating between both, I think it has no impact as long as we keep one format for the key (route.String() or route.Dst.String() )